scripts/github-actions/publish-az-types.js

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+* @typedef Params
+* @property {typeof require} require the 'require' function from github actions
+* @property {string} registryFqdn the fully qualified domain name of the registry to publish to
+* @property {string} tags the tags to apply to the published package
+* @property {typeof import("@actions/core")} core 
+*
+* @param {Params} params - The arguments for publishing the package.
+*
+* @returns {Promise<void>}
+*/
+async function publishTypeProviderPackage({ require, registryFqdn, tags, core }) {
```

Comment:
As a future improvement, I think it may be worth creating a command line tool (such as a .NET tool) that encapsulates the functionalities of this function (type bundling + publishing) and sharing it with other providers, so that they don't need to create their own CI workflow to bundle and publish types.

//cc @anthony-c-martin